test/EventTest.php

```diff
@@ -39,6 +39,7 @@ public function testGetParamReturnsDefault(): void
         $event   = new Event('foo', 'bar', []);
         $default = 1;
 
+        /** @psalm-suppress InvalidScalarArgument That's what we're testing. */
         self::assertEquals($default, $event->getParam('foo', $default));
```

Comment:
I think that the type additions to `getParam()` are  too strict here: what happens if we do a `getParam()` with an `object` as params? Will an `InvalidScalarArgument` be reported there too?

Reply:
To address both comments: I had to remove `key-of<TParams>` in 9f0e2eee872f9c03b00241db2221455b1b3fe5b0 because it only works on array types. `TParams` can be an object as well.

I've cleared the suppression because it was there to deal with this constraint (22957ca5f2a8548506de312000525cd7b7b5ccf7).